crates/matrix-sdk/src/widget/machine/tests/error.rs

```diff
@@ -189,3 +186,38 @@ fn send_request_for_non_allowed_message_like_events() {
     assert_eq!(msg["action"], "send_event");
     assert_eq!(msg["response"]["error"]["message"].as_str().unwrap(), "Not allowed");
 }
+
+#[test]
+fn read_request_for_message_like_with_msg_type_always_fails() {
+    // Note: this must fail since we don't have a way to construct a
+    // `ReadEventRequest` with a `msgtype` inside. In other words, we can't
+    // create such an input filter for the filtering engine that would match the
+    // filter that has `msgtype` in it.
```

Comment:
This isn't exactly correct, we can request more events than the widget is allowed to read, and only pass those back to the widget that it's allowed to read.

Reply:
I think the code is still fine as an initial implementation, we can adjust it in a follow-up PR. The comment should just be adjusted to say it's not *yet* supported in that case.

Reply:
Makes sense. Actually, there is no reason why a proper implementation cannot be part of this PR (it turns out it's not that much work). I've also added another integration test to check the behavior. Ptal!